Error when I running: "pip install -r cvat/requirements/development.txt"

I want change CVAT source, And I follow this: https://opencv.github.io/cvat/docs/contributing/development-environment/

util this step:

```shell
(.env) ming@ming:~/desktop/magic-version-cvat$ pip install -r cvat/requirements/development.txt
```

and Eror:

```shell
DEPRECATION: --no-binary currently disables reading from the cache of locally built wheels. In the future --no-binary will not influence the wheel cache. pip 23.1 will enforce this behaviour change. A possible replacement is to use the --no-cache-dir option. You can use the flag --use-feature=no-binary-enable-wheel-cache to test the upcoming behaviour. Discussion can be found at https://github.com/pypa/pip/issues/11453
Collecting attrs==21.4.0
  Using cached attrs-21.4.0-py2.py3-none-any.whl (60 kB)
Collecting click==7.1.2
  Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)
Collecting Django==3.2.15
  Using cached Django-3.2.15-py3-none-any.whl (7.9 MB)
Collecting django-appconf==1.0.4
  Using cached django_appconf-1.0.4-py2.py3-none-any.whl (6.3 kB)
Collecting django-auth-ldap==2.2.0
  Using cached django_auth_ldap-2.2.0-py3-none-any.whl (20 kB)
Collecting django-compressor==2.4
  Using cached django_compressor-2.4-py2.py3-none-any.whl (126 kB)
Collecting django-rq==2.3.2
  Using cached django_rq-2.3.2-py2.py3-none-any.whl (45 kB)
Requirement already satisfied: EasyProcess==0.3 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 8)) (0.3)
Collecting Pillow==9.0.1
  Using cached Pillow-9.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)
Collecting numpy==1.22.0
  Using cached numpy-1.22.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
Collecting python-ldap==3.4.0
  Using cached python_ldap-3.4.0-cp310-cp310-linux_x86_64.whl
Requirement already satisfied: pytz==2020.1 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 12)) (2020.1)
Collecting pyunpack==0.2.1
  Using cached pyunpack-0.2.1-py2.py3-none-any.whl (3.7 kB)
Requirement already satisfied: rcssmin==1.0.6 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 14)) (1.0.6)
Collecting redis==3.5.3
  Using cached redis-3.5.3-py2.py3-none-any.whl (72 kB)
Requirement already satisfied: rjsmin==1.1.0 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 16)) (1.1.0)
Collecting requests==2.26.0
  Using cached requests-2.26.0-py2.py3-none-any.whl (62 kB)
Collecting rq==1.5.1
  Using cached rq-1.5.1-py2.py3-none-any.whl (62 kB)
Collecting rq-scheduler==0.10.0
  Using cached rq_scheduler-0.10.0-py2.py3-none-any.whl (12 kB)
Collecting sqlparse==0.4.2
  Using cached sqlparse-0.4.2-py3-none-any.whl (42 kB)
Collecting django-sendfile2==0.7.0
  Using cached django_sendfile2-0.7.0-py3-none-any.whl (10 kB)
Requirement already satisfied: dj-pagination==2.5.0 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 22)) (2.5.0)
Collecting python-logstash-async==2.2.0
  Using cached python_logstash_async-2.2.0-py2.py3-none-any.whl (21 kB)
Requirement already satisfied: rules==2.2 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 24)) (2.2)
Collecting GitPython==3.1.27
  Using cached GitPython-3.1.27-py3-none-any.whl (181 kB)
Collecting coreapi==2.3.3
  Using cached coreapi-2.3.3-py2.py3-none-any.whl (25 kB)
Collecting django-filter==2.4.0
  Using cached django_filter-2.4.0-py3-none-any.whl (73 kB)
Collecting Markdown==3.2.2
  Using cached Markdown-3.2.2-py3-none-any.whl (88 kB)
Collecting djangorestframework==3.12.4
  Using cached djangorestframework-3.12.4-py3-none-any.whl (957 kB)
Collecting Pygments==2.7.4
  Using cached Pygments-2.7.4-py3-none-any.whl (950 kB)
Collecting drf-spectacular==0.22.1
  Using cached drf_spectacular-0.22.1-py3-none-any.whl (84 kB)
Requirement already satisfied: Shapely==1.7.1 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 32)) (1.7.1)
Collecting pdf2image==1.14.0
  Using cached pdf2image-1.14.0-py3-none-any.whl (10 kB)
Collecting dj-rest-auth[with_social]==2.2.4
  Using cached dj_rest_auth-2.2.4-py2.py3-none-any.whl
Collecting opencv-python-headless==4.5.5.62
  Using cached opencv_python_headless-4.5.5.62-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (47.7 MB)
Collecting h5py==3.6.0
  Using cached h5py-3.6.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (4.5 MB)
Collecting django-cors-headers==3.5.0
  Using cached django_cors_headers-3.5.0-py3-none-any.whl (11 kB)
Collecting furl==2.1.0
  Using cached furl-2.1.0-py2.py3-none-any.whl (20 kB)
Collecting av==8.0.2
  Using cached av-8.0.2.tar.gz (2.3 MB)
  Preparing metadata (setup.py) ... done
Collecting tensorflow==2.8.1
  Using cached tensorflow-2.8.1-cp310-cp310-manylinux2010_x86_64.whl (498.0 MB)
Requirement already satisfied: patool==1.12 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 43)) (1.12)
Requirement already satisfied: diskcache==5.0.2 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 44)) (5.0.2)
Collecting boto3==1.17.61
  Using cached boto3-1.17.61-py2.py3-none-any.whl (131 kB)
Collecting azure-storage-blob==12.13.0
  Using cached azure_storage_blob-12.13.0-py3-none-any.whl (377 kB)
Collecting google-cloud-storage==1.42.0
  Using cached google_cloud_storage-1.42.0-py2.py3-none-any.whl (105 kB)
Collecting datumaro==0.2.0
  Using cached datumaro-0.2.tar.gz (225 kB)
  Preparing metadata (setup.py) ... done
Collecting urllib3>=1.26.5
  Using cached urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
Collecting natsort==8.0.0
  Using cached natsort-8.0.0-py3-none-any.whl (37 kB)
Requirement already satisfied: mistune>=2.0.1 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/base.txt (line 53)) (2.0.4)
Collecting dnspython==2.2.0
  Using cached dnspython-2.2.0-py3-none-any.whl (266 kB)
Collecting pylint==2.14.5
  Using cached pylint-2.14.5-py3-none-any.whl (488 kB)
Collecting pylint-django==2.5.3
  Using cached pylint_django-2.5.3-py3-none-any.whl (81 kB)
Collecting pylint-plugin-utils==0.7
  Using cached pylint_plugin_utils-0.7-py3-none-any.whl (10 kB)
Requirement already satisfied: rope==0.17.0 in ./.env/lib/python3.10/site-packages (from -r cvat/requirements/development.txt (line 6)) (0.17.0)
Collecting django-extensions==3.0.8
  Using cached django_extensions-3.0.8-py3-none-any.whl (221 kB)
Collecting Werkzeug==1.0.1
  Using cached Werkzeug-1.0.1-py2.py3-none-any.whl (298 kB)
Collecting snakeviz==2.1.0
  Using cached snakeviz-2.1.0-py2.py3-none-any.whl (282 kB)
Collecting asgiref<4,>=3.3.2
  Using cached asgiref-3.5.2-py3-none-any.whl (22 kB)
Collecting six>=1.12.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Requirement already satisfied: pyasn1>=0.3.7 in ./.env/lib/python3.10/site-packages (from python-ldap==3.4.0->-r cvat/requirements/base.txt (line 11)) (0.4.8)
Collecting pyasn1-modules>=0.1.5
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Requirement already satisfied: entrypoint2 in ./.env/lib/python3.10/site-packages (from pyunpack==0.2.1->-r cvat/requirements/base.txt (line 13)) (1.1)
Collecting charset-normalizer~=2.0.0
  Using cached charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting croniter>=0.3.9
  Using cached croniter-1.3.7-py2.py3-none-any.whl (17 kB)
Collecting limits
  Using cached limits-2.7.1-py3-none-any.whl (41 kB)
Collecting pylogbeat
  Using cached pylogbeat-2.0.0-py2.py3-none-any.whl (10 kB)
Collecting gitdb<5,>=4.0.1
  Using cached gitdb-4.0.9-py3-none-any.whl (63 kB)
Requirement already satisfied: itypes in ./.env/lib/python3.10/site-packages (from coreapi==2.3.3->-r cvat/requirements/base.txt (line 26)) (1.2.0)
Collecting uritemplate
  Using cached uritemplate-4.1.1-py2.py3-none-any.whl (10 kB)
Collecting coreschema
  Using cached coreschema-0.0.4-py3-none-any.whl
Collecting inflection>=0.3.1
  Using cached inflection-0.5.1-py2.py3-none-any.whl (9.5 kB)
Collecting jsonschema>=2.6.0
  Using cached jsonschema-4.17.0-py3-none-any.whl (83 kB)
Collecting PyYAML>=5.1
  Using cached PyYAML-6.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (682 kB)
Collecting django-allauth<0.51.0,>=0.40.0
  Using cached django_allauth-0.50.0-py3-none-any.whl
Collecting orderedmultidict>=1.0.1
  Using cached orderedmultidict-1.0.1-py2.py3-none-any.whl (11 kB)
Collecting google-pasta>=0.1.1
  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)
Collecting opt-einsum>=2.3.2
  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting grpcio<2.0,>=1.24.3
  Using cached grpcio-1.50.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.7 MB)
Collecting astunparse>=1.6.0
  Using cached astunparse-1.6.3-py2.py3-none-any.whl (12 kB)
Collecting tensorflow-io-gcs-filesystem>=0.23.1
  Using cached tensorflow_io_gcs_filesystem-0.27.0-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (2.4 MB)
Requirement already satisfied: libclang>=9.0.1 in ./.env/lib/python3.10/site-packages (from tensorflow==2.8.1->-r cvat/requirements/base.txt (line 40)) (14.0.6)
Collecting typing-extensions>=3.6.6
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting wrapt>=1.11.0
  Using cached wrapt-1.14.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (77 kB)
Collecting absl-py>=0.4.0
  Using cached absl_py-1.3.0-py3-none-any.whl (124 kB)
Collecting gast>=0.2.1
  Using cached gast-0.5.3-py3-none-any.whl (19 kB)
Requirement already satisfied: keras<2.9,>=2.8.0rc0 in ./.env/lib/python3.10/site-packages (from tensorflow==2.8.1->-r cvat/requirements/base.txt (line 40)) (2.8.0)
Collecting keras-preprocessing>=1.1.1
  Using cached Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)
Collecting protobuf>=3.9.2
  Using cached protobuf-4.21.9-cp37-abi3-manylinux2014_x86_64.whl (408 kB)
Collecting tensorboard<2.9,>=2.8
  Using cached tensorboard-2.8.0-py3-none-any.whl (5.8 MB)
Requirement already satisfied: flatbuffers>=1.12 in ./.env/lib/python3.10/site-packages (from tensorflow==2.8.1->-r cvat/requirements/base.txt (line 40)) (22.10.26)
Requirement already satisfied: setuptools in ./.env/lib/python3.10/site-packages (from tensorflow==2.8.1->-r cvat/requirements/base.txt (line 40)) (65.5.1)
Collecting termcolor>=1.1.0
  Using cached termcolor-2.1.0-py3-none-any.whl (5.8 kB)
Requirement already satisfied: tensorflow-estimator<2.9,>=2.8 in ./.env/lib/python3.10/site-packages (from tensorflow==2.8.1->-r cvat/requirements/base.txt (line 40)) (2.8.0)
Collecting s3transfer<0.5.0,>=0.4.0
  Using cached s3transfer-0.4.2-py2.py3-none-any.whl (79 kB)
Collecting botocore<1.21.0,>=1.20.61
  Using cached botocore-1.20.112-py2.py3-none-any.whl (7.7 MB)
Collecting jmespath<1.0.0,>=0.7.1
  Using cached jmespath-0.10.0-py2.py3-none-any.whl (24 kB)
Collecting msrest>=0.6.21
  Using cached msrest-0.7.1-py3-none-any.whl (85 kB)
Collecting azure-core<2.0.0,>=1.23.1
  Using cached azure_core-1.26.1-py3-none-any.whl (172 kB)
Collecting cryptography>=2.1.4
  Using cached cryptography-38.0.3-cp36-abi3-manylinux_2_28_x86_64.whl (4.2 MB)
Collecting google-auth<3.0dev,>=1.25.0
  Using cached google_auth-2.14.1-py2.py3-none-any.whl (175 kB)
Collecting google-api-core<3.0dev,>=1.29.0
  Using cached google_api_core-2.10.2-py3-none-any.whl (115 kB)
Collecting google-resumable-media<3.0dev,>=1.3.0
  Using cached google_resumable_media-2.4.0-py2.py3-none-any.whl (77 kB)
Collecting google-cloud-core<3.0dev,>=1.6.0
  Using cached google_cloud_core-2.3.2-py2.py3-none-any.whl (29 kB)
Collecting defusedxml>=0.7.0
  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting lxml>=4.4.1
  Using cached lxml-4.9.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (6.9 MB)
Collecting matplotlib>=3.3.1
  Using cached matplotlib-3.6.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.8 MB)
Collecting networkx>=2.5
  Using cached networkx-2.8.8-py3-none-any.whl (2.0 MB)
Collecting ruamel.yaml>=0.17.0
  Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
Collecting tensorboardX!=2.3,>=1.8
  Using cached tensorboardX-2.5.1-py2.py3-none-any.whl (125 kB)
Collecting scipy
  Using cached scipy-1.9.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (33.7 MB)
Collecting pandas>=1.1.5
  Using cached pandas-1.5.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.1 MB)
Collecting opencv-python
  Using cached opencv_python-4.6.0.66-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (60.9 MB)
Collecting pycocotools!=2.0.2,>=2.0.0
  Using cached pycocotools-2.0.6-cp310-cp310-linux_x86_64.whl
Collecting mccabe<0.8,>=0.6
  Using cached mccabe-0.7.0-py2.py3-none-any.whl (7.3 kB)
Collecting dill>=0.2
  Using cached dill-0.3.6-py3-none-any.whl (110 kB)
Collecting platformdirs>=2.2.0
  Using cached platformdirs-2.5.3-py3-none-any.whl (14 kB)
Collecting tomli>=1.1.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting isort<6,>=4.2.5
  Using cached isort-5.10.1-py3-none-any.whl (103 kB)
Collecting tomlkit>=0.10.1
  Using cached tomlkit-0.11.6-py3-none-any.whl (35 kB)
Collecting astroid<=2.12.0-dev0,>=2.11.6
  Using cached astroid-2.11.7-py3-none-any.whl (251 kB)
Collecting tornado>=2.0
  Using cached tornado-6.2-cp37-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (423 kB)
Collecting lazy-object-proxy>=1.4.0
  Using cached lazy_object_proxy-1.8.0-cp310-cp310-linux_x86_64.whl
Requirement already satisfied: wheel<1.0,>=0.23.0 in ./.env/lib/python3.10/site-packages (from astunparse>=1.6.0->tensorflow==2.8.1->-r cvat/requirements/base.txt (line 40)) (0.38.3)
Collecting python-dateutil<3.0.0,>=2.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting cffi>=1.12
  Using cached cffi-1.15.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (441 kB)
Collecting pyjwt[crypto]>=1.7
  Using cached PyJWT-2.6.0-py3-none-any.whl (20 kB)
Collecting requests-oauthlib>=0.3.0
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting python3-openid>=3.0.8
  Using cached python3_openid-3.2.0-py3-none-any.whl (133 kB)
Collecting smmap<6,>=3.0.1
  Using cached smmap-5.0.0-py3-none-any.whl (24 kB)
Collecting googleapis-common-protos<2.0dev,>=1.56.2
  Using cached googleapis_common_protos-1.56.4-py2.py3-none-any.whl (211 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.2.0-py3-none-any.whl (9.3 kB)
Collecting google-crc32c<2.0dev,>=1.0
  Using cached google_crc32c-1.5.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (32 kB)
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Using cached pyrsistent-0.19.2-py3-none-any.whl (57 kB)
Collecting contourpy>=1.0.1
  Using cached contourpy-1.0.6-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (296 kB)
Collecting pyparsing>=2.2.1
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.38.0-py3-none-any.whl (965 kB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.4.4-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.6 MB)
Collecting isodate>=0.6.0
  Using cached isodate-0.6.1-py2.py3-none-any.whl (41 kB)
Collecting ruamel.yaml.clib>=0.2.6
  Using cached ruamel.yaml.clib-0.2.7-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (485 kB)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in ./.env/lib/python3.10/site-packages (from tensorboard<2.9,>=2.8->tensorflow==2.8.1->-r cvat/requirements/base.txt (line 40)) (1.8.1)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-manylinux2010_x86_64.whl (4.9 MB)
Collecting protobuf>=3.9.2
  Using cached protobuf-3.19.6-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
Collecting jinja2
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting deprecated>=1.2
  Using cached Deprecated-1.2.13-py2.py3-none-any.whl (9.6 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Installing collected packages: av, wrapt, Werkzeug, urllib3, uritemplate, typing-extensions, tornado, tomlkit, tomli, termcolor, tensorflow-io-gcs-filesystem, tensorboard-data-server, sqlparse, smmap, six, ruamel.yaml.clib, rsa, redis, PyYAML, pyunpack, pyrsistent, pyparsing, pylogbeat, pyjwt, Pygments, pycparser, pyasn1-modules, protobuf, platformdirs, Pillow, oauthlib, numpy, networkx, natsort, mccabe, MarkupSafe, Markdown, lxml, lazy-object-proxy, kiwisolver, jmespath, isort, inflection, idna, google-crc32c, gast, fonttools, dnspython, django-extensions, dill, defusedxml, cycler, click, charset-normalizer, certifi, cachetools, attrs, asgiref, absl-py, tensorboardX, snakeviz, scipy, ruamel.yaml, rq, requests, python3-openid, python-ldap, python-dateutil, pdf2image, packaging, orderedmultidict, opt-einsum, opencv-python-headless, opencv-python, keras-preprocessing, jsonschema, jinja2, isodate, h5py, grpcio, googleapis-common-protos, google-resumable-media, google-pasta, google-auth, gitdb, Django, deprecated, contourpy, cffi, astunparse, astroid, requests-oauthlib, pylint, pandas, matplotlib, limits, google-api-core, GitPython, furl, djangorestframework, django-sendfile2, django-rq, django-filter, django-cors-headers, django-auth-ldap, django-appconf, cryptography, croniter, coreschema, botocore, azure-core, s3transfer, rq-scheduler, python-logstash-async, pylint-plugin-utils, pycocotools, msrest, google-cloud-core, google-auth-oauthlib, drf-spectacular, django-compressor, dj-rest-auth, coreapi, tensorboard, pylint-django, google-cloud-storage, django-allauth, datumaro, boto3, azure-storage-blob, tensorflow
  DEPRECATION: av is being installed using the legacy 'setup.py install' method, because the '--no-binary' option was enabled for it and this currently disables local wheel building for projects that don't have a 'pyproject.toml' file. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/11451
  Running setup.py install for av ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for av did not run successfully.
  │ exit code: 1
  ╰─> [57 lines of output]
      running install
      /home/ming/桌面/magic-version-cvat/.env/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-310
      creating build/lib.linux-x86_64-cpython-310/av
      copying av/datasets.py -> build/lib.linux-x86_64-cpython-310/av
      copying av/__main__.py -> build/lib.linux-x86_64-cpython-310/av
      copying av/deprecation.py -> build/lib.linux-x86_64-cpython-310/av
      copying av/__init__.py -> build/lib.linux-x86_64-cpython-310/av
      creating build/lib.linux-x86_64-cpython-310/av/data
      copying av/data/__init__.py -> build/lib.linux-x86_64-cpython-310/av/data
      creating build/lib.linux-x86_64-cpython-310/av/sidedata
      copying av/sidedata/__init__.py -> build/lib.linux-x86_64-cpython-310/av/sidedata
      creating build/lib.linux-x86_64-cpython-310/av/codec
      copying av/codec/__init__.py -> build/lib.linux-x86_64-cpython-310/av/codec
      creating build/lib.linux-x86_64-cpython-310/av/subtitles
      copying av/subtitles/__init__.py -> build/lib.linux-x86_64-cpython-310/av/subtitles
      creating build/lib.linux-x86_64-cpython-310/av/container
      copying av/container/__init__.py -> build/lib.linux-x86_64-cpython-310/av/container
      creating build/lib.linux-x86_64-cpython-310/av/video
      copying av/video/__init__.py -> build/lib.linux-x86_64-cpython-310/av/video
      creating build/lib.linux-x86_64-cpython-310/av/filter
      copying av/filter/__init__.py -> build/lib.linux-x86_64-cpython-310/av/filter
      creating build/lib.linux-x86_64-cpython-310/av/audio
      copying av/audio/__init__.py -> build/lib.linux-x86_64-cpython-310/av/audio
      running build_ext
      running config
      PyAV: 8.0.2 (unknown commit)
      Python: 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0]
      platform: Linux-5.15.0-52-generic-x86_64-with-glibc2.35
      extension_extra:
          include_dirs: [b'include', b'/usr/include/x86_64-linux-gnu']
          libraries: [b'avformat', b'avcodec', b'avdevice', b'avutil', b'avfilter', b'swscale', b'swresample']
          library_dirs: []
          define_macros: []
          runtime_library_dirs: []
      config_macros:
          PYAV_COMMIT_STR="unknown-commit"
          PYAV_VERSION=8.0.2
          PYAV_VERSION_STR="8.0.2"
      writing build/temp.linux-x86_64-cpython-310/include/pyav/config.h
      running cythonize
      building 'av.bytesource' extension
      creating build/temp.linux-x86_64-cpython-310/src
      creating build/temp.linux-x86_64-cpython-310/src/av
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/usr/include/x86_64-linux-gnu -Iinclude -I/home/ming/桌面/magic-version-cvat/.env/include -Ibuild/temp.linux-x86_64-cpython-310/include -I/usr/include/python3.10 -I/home/ming/桌面/magic-version-cvat/.env/include -I/usr/include/python3.10 -Ibuild/temp.linux-x86_64-cpython-310/include -c src/av/bytesource.c -o build/temp.linux-x86_64-cpython-310/src/av/bytesource.o
      src/av/bytesource.c: In function ‘__pyx_tp_dealloc_2av_10bytesource_ByteSource’:
      src/av/bytesource.c:1871:5: error: lvalue required as increment operand
       1871 |     ++Py_REFCNT(o);
            |     ^~
      src/av/bytesource.c:1873:5: error: lvalue required as decrement operand
       1873 |     --Py_REFCNT(o);
            |     ^~
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> av

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

```





My Envirenment:

System: Ubuntu 22.04 (LTS)

Python version: 3.10.6

Pip version: 22.3.1

--- a/cvat/requirements/base.txt
+++ b/cvat/requirements/base.txt
@@ -15,7 +15,7 @@ rcssmin==1.0.6
 redis==3.5.3
 rjsmin==1.1.0
 requests==2.26.0
-rq==1.5.1
+rq==1.5.2
 rq-scheduler==0.10.0
 sqlparse==0.4.2
 django-sendfile2==0.7.0
@@ -36,7 +36,7 @@ opencv-python-headless==4.5.5.62
 h5py==3.6.0
 django-cors-headers==3.5.0
 furl==2.1.0
-av==8.0.2 --no-binary=av
+av==9.2.0 --no-binary=av
 tensorflow==2.9.3 # Optional requirement of Datumaro. Use tensorflow-macos==2.8.0 for Mac M1
 # The package is used by pyunpack as a command line tool to support multiple
 # archives. Don't use as a python module because it has GPL license.

--- a/utils/dataset_manifest/requirements.txt
+++ b/utils/dataset_manifest/requirements.txt
@@ -1,4 +1,4 @@
-av==8.0.2 --no-binary=av
+av==9.2.0 --no-binary=av
 opencv-python-headless==4.4.0.42
 Pillow==9.3.0
 tqdm==4.58.0

